### DISCUSSION TITLE
ESP32 + LittleFS. How to access a file included by manifest.

### DISCUSSION THREAD

**Discussion by AlexLucardGoodman** [Moddable-OpenSource/moddable#963] [General]
Hello!
I'm experimenting with LittleFS filesystem on Esp32. I've been able to read directory's contents, create and delete files, but I was wondering how to access a file included in the Moddable Project in my manifest.json???

During some WifiAP tests I created a server and was able to send a file included as follows in manifest: 
`
...
"data": {
		"*": [
			"./assets/textfile"
		]
	}
...
`
but this file doesn't appear in the filesystem. Is there a way to access it through LittleFS?
Thank you.

**Comment by phoddie** [comment]
The files referenced by the `data` section of the manifest are transferred to the device as [Resources](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/files/files.md#resource). They do not build a file system. Resources are lighter-weight than files and work even on all devices, even those without a file system. You access the data in a resource as a read-only `Uint8Array`:

```js
import Resource from "Resource";

let bytes = new Uint8Array(new Resource("mydata"));
```

The [files/resource](https://github.com/Moddable-OpenSource/moddable/blob/public/examples/files/resource/main.js) example goes a bit deeper.

If you want to make the resource data available through a file system, you can write some code to copy the data from the resource to a file.
